plugin/dnstap/handler.go

```diff
@@ -32,14 +31,6 @@ type (
 	}
 )
 
-// ContextKey defines the type of key that is used to save data into the context.
-type ContextKey string
-
-const (
-	// DnstapSendOption specifies the Dnstap message to be send.  Default is sent all.
-	DnstapSendOption ContextKey = "dnstap-send-option"
```

Comment:
the plugin only has one option `full`, for the rest this was set to true. So I that flexibility is needed (it never was) config options should be added. 
Per previous comment the whole context stuff is gone, so this can be this be a bunch of booleans in `Dnstap`

Reply:
Yes, these options are really needed. AFAIK bind and unbound allow configuring per message type as well.
The config may look like
```
  dnstap tcp://127.0.0.1:9123 full CR FR
```
where all parameters after endpoint are optional
or
```
  dnstap tcp://127.0.0.1:9123 full {
    allow CR FR
  }
```
Other plugins should be able to check these config options. Bunch of booleans is ok.